iree/compiler/Dialect/Flow/Transforms/LegalizeInputTypes.cpp

```diff
@@ -85,7 +85,8 @@ LogicalResult convertOperation(Operation *oldOp,
                                FlowTypeConverter &typeConverter,
                                BlockAndValueMapping &mapping,
                                OpBuilder &builder) {
-  if (llvm::isa<mlir::linalg::LinalgOp>(oldOp)) {
+  if (llvm::isa<mlir::linalg::LinalgOp>(oldOp) ||
+      llvm::isa<mlir::linalg::YieldOp>(oldOp)) {
     // Currently assumes linalg ops have legal types.
     // TODO: rewrite to generic and back.
```

Comment:
I don't really understand the TODO here, and you're adding another case to the condition that it comments on. Can you expand on the TODO?

This now also assumes that YieldOps are already legal? What would "rewriting to generic and back" accomplish?

Reply:
I updated to check directly that we assume all linalg ops are valid (I think this is what the original code was meant to do). Tried to update the comment but the original line was a bit of a mystery to me too.